packages/hooks/src/useP2PAdvertList.ts

```diff
@@ -1,15 +1,20 @@
 import React from 'react';
-import { usePaginatedFetch } from '@deriv/api';
+import { useFetch } from '@deriv/api';
 import useExchangeRate from './useExchangeRate';
 
 /**
  * This custom hook returns available adverts for use with 'p2p_order_create' by calling 'p2p_advert_list' endpoint
  */
-const useP2PAdvertList = (
-    payload?: NonNullable<Parameters<typeof usePaginatedFetch<'p2p_advert_list'>>[1]>['payload']
-) => {
+const useP2PAdvertList = (payload?: NonNullable<Parameters<typeof useFetch<'p2p_advert_list'>>[1]>['payload']) => {
     const { getRate } = useExchangeRate();
-    const { data, ...rest } = usePaginatedFetch('p2p_advert_list', { payload });
+
+    const limit: number = payload?.limit || 10;
```

Comment:
do we need to have it default to 10 if theres no limit passing? because when we call the list, if we dont pass limit its defaulted to 50 already

Reply:
@ameerul-deriv, it's safer to provide a fallback value here, since the values in the `payload` are all optional and possibly `undefined` . It'd be dangerous in my opinion to typecast `Number(undefined)`

Reply:
I'll just bump the limit from `10` to `50`, to follow the `p2p_advert_list` api

Reply:
yeah thats fine as well 👍